### PR TITLE
use float instead of double

### PR DESCRIPTION
**Pull request by InternetofAwesome**

### CODE REVIEW

File: libc/baselibc/src/tinyprintf.c [hunk 1]
```diff
@@ -356,7 +356,7 @@ size_t tfp_format(FILE *putp, const char *fmt, va_list va)
 #if MYNEWT_VAL(FLOAT_USER)
             case 'f':
                 p.base = 10;
-                d = va_arg(va, double);
+                d = (float)va_arg(va, double);
```
[the remark this answers is not in the document]
Comment (andrzej-kaczmarek) [reply libc/baselibc/src/tinyprintf.c]: `float` is always promoted to `double` when passed  as variable argument so reading `float` will return invalid value

[the remark this answers is not in the document]
Comment (InternetofAwesome) [reply libc/baselibc/src/tinyprintf.c]: Yeah, I tried that first, and looked into a workaround. Unfortunately, without doing some pretty nasty hacking, it wasn't feasible.

**Comment by ccollins476ad** [comment]
Does this actually allow for any optimizations?  As @andrzej-kaczmarek said, floats are always promoted to doubles anyway, so I am skeptical that this actually does anything (but then what do I know?  I am not a compiler...).

If this is a useful change, then it should be conditional on a syscfg setting.  `%f` is defined as the double format specifier in the C spec, so that should be the default behavior.

**Comment by andrzej-kaczmarek** [comment]
@ccollins476ad since e.g. Cortex-M4 has (optional) FPU which supports single precision operations, on those MCUs `double` will be converted to `float` in sw and then calculations will be done by FPU so there will be some performance gain here. And I agree that this should be conditional on syscfg setting so we do not break standard in default configuration.

**Comment by ccollins476ad** [comment]
Thanks, @andrzej-kaczmarek, I wasn't aware of that.  Still... does the FPU help in converting floating point numbers to strings?

**Comment by ccollins476ad** [comment]
Closing this as development seems to have stalled.  Please reopen if this is still being worked on.

**Comment by InternetofAwesome** [comment]
This issue should definitely be addressed one way or the other. As the code exists right now, it pulls in a few unnecessary double software functions that would not be compiled in with these changes.
That said, this code was not written with any use case other than my own in mind - For the use case of having a single-precision FPU, this is appropriate, as the only double function that gets compiled in as a result is the function to convert double to float.

**Pull request closed without merging**